packages/babel-traverse/src/path/replacement.js

```diff
@@ -183,15 +183,15 @@ export function _replaceWith(node) {
     throw new ReferenceError("Container is falsy");
   }
 
+  this.debug(() => `Replace with ${node && node.type}`);
+
+  this.node = this.container[this.key] = node;
```

Comment:
Shouldn't we first validate and then set then replace? Or what is the idea of this change?